### PR TITLE
improve smbmount another time

### PR DESCRIPTION
**Pull request by szaimen**

### CODE REVIEW

File: apps/smbmount.sh [hunk 1]
```diff
@@ -123,23 +131,35 @@ do
...
+            if [ -f $SMB_CREDENTIALS/SMB$count ]
+            then
+                check_command rm $SMB_CREDENTIALS/SMB$count
+            fi
             break
         else
-            # Install and enable files_external
-            if ! is_app_enabled files_external
+            # Check if Nextcloud is existing
+            if [ -f $NCPATH/occ ]
```
Comment (enoch85) [inline apps/smbmount.sh:141]: Maybe add this to all the functions tha use 'occ' instead?

**Comment by enoch85** [comment]
Nice work!

File: apps/smbmount.sh (hunk 1 above)
Comment (szaimen) [reply apps/smbmount.sh:141]: What do you mean by that?

**Comment by szaimen** [comment]
> Nice work!

Thanks!
It is already tested. I am only waiting for travis if there are some suggestions...

**Comment by szaimen** [comment]
travis is fixed

**Comment by enoch85** [comment]
Is this done?

File: apps/smbmount.sh (hunk 1 above)
Comment (enoch85) [reply apps/smbmount.sh:141]: add:

```
if [ -f $NCPATH/occ ]
then
    some_occ_command
fi
```

...in each function that use `occ`

Comment (szaimen) [reply apps/smbmount.sh:141]: So shall I add this to the occ_command or occ_command_no_check functions?
Is that your suggestion?

concerning the script: the only places where I use them is already covered by this one check.

**Comment by szaimen** [comment]
> Is this done?

yes, we could only debate, whether we can remove or change those lines now:
https://github.com/nextcloud/vm/blob/bc5a76c0ce3d1b4584e305f5bdc0a9f747291aa7/apps/smbmount.sh#L41-L49

**Comment by enoch85** [comment]
>  remove or change those lines now

Since you now seem to use `.smbcredentials` those lines can be removed. 

`.smbcredentials` are the file that should be secured as there's no important stuff in `fstab`.

**Comment by szaimen** [comment]
> > remove or change those lines now
> 
> Since you now seem to use `.smbcredentials` those lines can be removed.
> 
> `.smbcredentials` are the file that should be secured as there's no important stuff in `fstab`.

okay, then I remove those.
and yes, the files in .smbcredentials are secured.

**Comment by szaimen** [comment]
this is done 👍

File: apps/smbmount.sh (hunk 1 above)
Comment (szaimen) [reply apps/smbmount.sh:131]: @enoch85 I need an answer here before merging..

**Pull request merged**